pkg/services/apiserver/client/client_test.go

```diff
@@ -9,26 +9,41 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestGetUserFromMeta(t *testing.T) {
+func TestGetUsersFromMeta(t *testing.T) {
 	userSvcTest := usertest.NewUserServiceFake()
-	userSvcTest.ExpectedUser = &user.User{
-		ID:  1,
-		UID: "uid-value",
+	userSvcTest.ExpectedListUsersByIdOrUid = []*user.User{
+		{
+			ID:  1,
+			UID: "uid-value",
+		},
+		{
+			ID:  2,
+			UID: "uid-value2",
+		},
 	}
 	client := &k8sHandler{
 		userService: userSvcTest,
 	}
 	t.Run("returns user with valid UID", func(t *testing.T) {
-		result, err := client.GetUserFromMeta(context.Background(), "user:uid-value")
+		result, err := client.GetUsersFromMeta(context.Background(), []string{"user:uid-value"})
 		require.NoError(t, err)
-		require.Equal(t, "uid-value", result.UID)
-		require.Equal(t, int64(1), result.ID)
+		require.Equal(t, "uid-value", result["user:uid-value"].UID)
+		require.Equal(t, int64(1), result["user:uid-value"].ID)
 	})
 
 	t.Run("returns user when id is passed in", func(t *testing.T) {
-		result, err := client.GetUserFromMeta(context.Background(), "user:1")
+		result, err := client.GetUsersFromMeta(context.Background(), []string{"user:1"})
 		require.NoError(t, err)
-		require.Equal(t, "uid-value", result.UID)
-		require.Equal(t, int64(1), result.ID)
+		require.Equal(t, "uid-value", result["user:1"].UID)
+		require.Equal(t, int64(1), result["user:1"].ID)
+	})
+
+	t.Run("returns users when id and uid are passed in", func(t *testing.T) {
+		result, err := client.GetUsersFromMeta(context.Background(), []string{"user:1", "user:uid-value2"})
+		require.NoError(t, err)
+		require.Equal(t, "uid-value", result["user:1"].UID)
+		require.Equal(t, int64(1), result["user:1"].ID)
+		require.Equal(t, "uid-value2", result["user:uid-value2"].UID)
+		require.Equal(t, int64(2), result["user:uid-value2"].ID)
```

Comment:
is this correct?

Reply:
Should be - we want the mapping to be user metadata -> the user. What's your concern with it?

Reply:
scratch that i missed the source data. i thought it was being parsed